pymongo/asynchronous/mongo_client.py

```diff
@@ -1565,6 +1565,8 @@ async def close(self) -> None:
             # TODO: PYTHON-1921 Encrypted MongoClients cannot be re-opened.
             await self._encrypter.close()
         self._closed = True
+        # Yield to the asyncio event loop so all executor tasks properly exit after cancellation
+        await asyncio.sleep(0)
```

Comment:
Yielding here doesn't guarantee anything about task cleanup. Shouldn't we actually await the tasks to ensure they are cleaned up properly?